### PR TITLE
Cool Things

### PR DESCRIPTION
**Pull request by 77seven77seven**
## About the Pull Request

a big PR that adds the stuff from TG, mostly traitor items and QoL features

## Changelog

:cl:
rscadd: You can now put pens into paper bins, save space on your table, today!
rscadd: The Powerfist - this gas-powered fist adds a BIG boost to your punch, just make sure you have the necessary equipment to use it properly! (Gas not included.)
rscadd: The Radioactive Microlaser - a gadget cleverly disguised as a health analyzer, use it on someone and instead of checking their health, make it worse by adding a few millisieverts. Probably not the best idea to use this without proper protection.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/powerfist.dm [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+			user.put_in_hands(tank)
+			to_chat(user, SPAN_NOTICE("You detach \the [tank] from \the [src]."))
+			tank = null
+		else
+			to_chat(user, SPAN_WARNING("No tank present!"))
+
+	if(isWrench(W))
+		playsound(src.loc, 'sound/items/Ratchet.ogg', 50, 1)
+		fist_pressure_setting = fist_pressure_setting >= HIGH_PRESSURE ? LOW_PRESSURE : fist_pressure_setting + 1
+		to_chat(user, SPAN_NOTICE("Piston strength set to [pressure_setting_to_text(fist_pressure_setting)]!"))
```
Comment (EgorDinamit) [inline code/game/objects/items/weapons/powerfist.dm]: Add returns after the end of if() statements.

File: code/game/objects/items/weapons/powerfist.dm [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		if(tank)
+			var/affecting = user.get_organ_target()
+			if(tank.air_contents.volume >= gas_per_fist*fist_pressure_setting)
+				tank.air_contents.remove_volume(gas_per_fist*fist_pressure_setting)
+				src.visible_message(SPAN_DANGER("[user] punches \the [M] with \the [src]!"))
+				M.apply_damage(force*2*fist_pressure_setting, BRUTE, affecting) //might be a little too op... Too bad!
+			else
+				src.visible_message(SPAN_WARNING("\The [src] lets out a dull hiss..."))
+		else
+			to_chat(user, SPAN_WARNING("\The [src] doesn't have a tank!"))
```
Comment (EgorDinamit) [inline code/game/objects/items/weapons/powerfist.dm]: Utilize early returns instead of long if-else trees.

File: code/modules/mob/inventory.dm [hunk 3]
```diff
@@ -332,3 +332,7 @@ var/list/slot_equipment_priority = list( \
 		var/obj/item/I = entry
 		if(I.body_parts_covered & body_parts)
 			. += I
+
+//not sure if this is duplicate but i couldnt find a proc that does this so temporary until i get called out for being an idiot i suppose
+/mob/proc/transferItemToLoc(obj/item/I, newloc = null, force = FALSE, silent = TRUE)
+	. = unEquip(I, newloc)
```
Comment (EgorDinamit) [inline code/modules/mob/inventory.dm]: But all it does is just... call the proc that already exists..?

File: code/modules/paperwork/paperbin.dm [hunk 4]
```diff
@@ -101,10 +126,16 @@
 			to_chat(user, "<span class='notice'>There " + (amount > 1 ? "are [amount] papers" : "is one paper") + " in the bin.</span>")
 		else
 			to_chat(user, "<span class='notice'>There are no papers in the bin.</span>")
+	if(bin_pen)
+		to_chat(user, "There is \a [bin_pen] is in \the [src].")
```
Comment (EgorDinamit) [inline code/modules/paperwork/paperbin.dm]:
```suggestion
		to_chat(user, "There is \a [bin_pen] in \the [src].")
```

File: code/modules/mob/inventory.dm (hunk 3 above)
Comment (77seven77seven) [reply code/modules/mob/inventory.dm]: just as the comment foretold!!

File: code/game/objects/items/weapons/powerfist.dm (hunk 2 above)
Comment (77seven77seven) [reply code/game/objects/items/weapons/powerfist.dm]: example pls?

**Comment by 77seven77seven** [comment]
looks like this is the same branch in which i added the incendiary bullets, i should probably make them available for traitors, no?

File: code/modules/mob/inventory.dm (hunk 3 above)
Comment (77seven77seven) [reply code/modules/mob/inventory.dm]: fixed

File: code/game/objects/items/weapons/powerfist.dm (hunk 1 above)
Comment (77seven77seven) [reply code/game/objects/items/weapons/powerfist.dm]: done

File: code/game/objects/items/weapons/powerfist.dm (hunk 2 above)
Comment (77seven77seven) [reply code/game/objects/items/weapons/powerfist.dm]: fixed.

**Comment by 77seven77seven** [comment]
guess this is ready

**Review by EgorDinamit — APPROVED** [review]
There's honestly quite a lot of things that I would've done differently, but this should work, hopefuly...

**Pull request merged**